src/group_theory/subgroup.lean

```diff
@@ -1406,6 +1406,17 @@ le_antisymm
     (gclosure_preimage_le _ _))
   ((closure_le _).2 $ set.image_subset _ subset_closure)
 
+-- this instance can't go just after the definition of mrange because `fintype` is
+-- not imported at that stage
+@[to_additive]
+instance fintype_mrange {M : Type*} [monoid M] {N : Type*} [monoid N] [decidable_eq N]
+  (f : M →* N) [fintype M] : fintype (mrange f) :=
+set.fintype_range f
+
+@[to_additive]
+instance fintype_range [decidable_eq N] (f : G →* N) [fintype G] : fintype (range f) :=
```

Comment:
Usually the typeclasses go before the variables:
```suggestion
instance fintype_range  [fintype G] [decidable_eq N] (f : G →* N) : fintype (range f) :=
```
(same throughout)

I think I remember some weirdness with some lemmas / defs that didn't follow this rule.